### PR TITLE
[W5.4][T17-A3]Jeremy Long

### PR DESCRIPTION
**Pull request by Jeremylsw**

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -49,6 +49,23 @@ Examples:
 * `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
 * `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
 
+### Editing Employment status 
+Edits a person's employment status in the address book<br>
+Format: edites NAME
```
Comment (b02b01) [inline doc/UserGuide.md]: 'edites' should be 'edits'

File: doc/UserGuide.md [hunk 2]
```diff
@@ -49,6 +49,23 @@ Examples:
...
+
+> Person needs to be in a registered contact in the addressbook for the command to implement successfully
+Example: 
+* `edites jeremy
```
Comment (b02b01) [inline doc/UserGuide.md]: 'edites' should be 'edits'

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -128,4 +128,32 @@ public boolean equals(Object other) {
...
+    public void editEmploymentInfo(String name){
+        int editCounter=0;
+        for (Person p : allPersons){
+            if (p.getName().toString().equals(name.substring(1))){
+                p.editEmploymentInfo();
+                editCounter=1;
+            }
+            if (editCounter==1){return;}
+        }
+        System.out.println("|| Person not found! Please ensure person is a registered contact" +
```
Comment (b02b01) [inline src/seedu/addressbook/data/AddressBook.java]: Should use the `LINE_PREFIX` constant in `TextUi.java`

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -128,4 +128,32 @@ public boolean equals(Object other) {
...
+                " before using employment status function!");
+    }
+
+    public void viewEmploymentInfo(String name){
+        int viewedCounter=0;
```
Comment (b02b01) [inline src/seedu/addressbook/data/AddressBook.java]: Coding style issue: `int viewedCounter = 0`

File: src/seedu/addressbook/data/AddressBook.java [hunk 5]
```diff
@@ -128,4 +128,32 @@ public boolean equals(Object other) {
...
+        for (Person p: allPersons){
+            if (p.getName().toString().equals(name.substring(1))){
```
Comment (b02b01) [inline src/seedu/addressbook/data/AddressBook.java]: Require a space before `{`

File: src/seedu/addressbook/data/AddressBook.java [hunk 6]
```diff
@@ -128,4 +128,32 @@ public boolean equals(Object other) {
                         && this.allPersons.equals(((AddressBook) other).allPersons)
                         && this.allTags.equals(((AddressBook) other).allTags));
     }
+
+    public void editEmploymentInfo(String name){
+        int editCounter=0;
+        for (Person p : allPersons){
+            if (p.getName().toString().equals(name.substring(1))){
```
Comment (b02b01) [inline src/seedu/addressbook/data/AddressBook.java]: Should create a variable `personToEdit = name.substring(1)` so that the `.substring()` method does not have to be called in every iteration

Comment (Jeremylsw) [reply src/seedu/addressbook/data/AddressBook.java]: Edited and pushed

File: src/seedu/addressbook/data/AddressBook.java (hunk 3 above)
Comment (Jeremylsw) [reply src/seedu/addressbook/data/AddressBook.java]: Edited and pushed

File: src/seedu/addressbook/data/AddressBook.java (hunk 4 above)
Comment (Jeremylsw) [reply src/seedu/addressbook/data/AddressBook.java]: Reformatted all files. Committed and pushed.

File: src/seedu/addressbook/data/AddressBook.java (hunk 5 above)
Comment (Jeremylsw) [reply src/seedu/addressbook/data/AddressBook.java]: Reformatted all files using inbuilt functionality. Committed and pushed.

File: src/seedu/addressbook/Main.java [hunk 7]
```diff
@@ -22,22 +22,28 @@
  */
 public class Main {
 
-    /** Version info of the program. */
+    /**
```
Comment (liuyang1213) [inline src/seedu/addressbook/Main.java]: Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use sourcetree to look through changes before committing, and discard any unrelated changes.

File: src/seedu/addressbook/commands/EditEmploymentStatusCommand.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+    String name;
+
+    public EditEmploymentStatusCommand(String name) {
+        this.name = name;
+    }
+
+    public CommandResult execute() {
+        super.editEmploymentInfo(name);
+        return new CommandResult("Please enter next command");
```
Comment (liuyang1213) [inline src/seedu/addressbook/commands/EditEmploymentStatusCommand.java]: Should you use the raw string here?

File: src/seedu/addressbook/data/AddressBook.java [hunk 9]
```diff
@@ -125,7 +129,39 @@ public UniqueTagList getAllTags() {
...
+
+    public void editEmploymentInfo(String name) {
+        int editCounter = 0;
+        String newName = name.substring(1);
+        for (Person p : allPersons) {
+            if (p.getName().toString().equals(newName)) {
+                p.editEmploymentInfo();
+                editCounter = 1;
+            }
+            if (editCounter == 1) {
```
Comment (liuyang1213) [inline src/seedu/addressbook/data/AddressBook.java]: You could just put `return` statement right after `editCounter = 1;` Same for the code in `viewEmploymentInfo`

File: src/seedu/addressbook/data/AddressBook.java [hunk 10]
```diff
@@ -125,7 +129,39 @@ public UniqueTagList getAllTags() {
...
                 || (other instanceof AddressBook // instanceof handles nulls
-                        && this.allPersons.equals(((AddressBook) other).allPersons)
-                        && this.allTags.equals(((AddressBook) other).allTags));
+                && this.allPersons.equals(((AddressBook) other).allPersons)
+                && this.allTags.equals(((AddressBook) other).allTags));
+    }
+
+    public void editEmploymentInfo(String name) {
+        int editCounter = 0;
+        String newName = name.substring(1);
```
Comment (liuyang1213) [inline src/seedu/addressbook/data/AddressBook.java]: Try to avoid magic number.

File: src/seedu/addressbook/commands/Command.java [hunk 11]
```diff
@@ -67,4 +71,12 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public void editEmploymentInfo(String name) {
```
Comment (liuyang1213) [inline src/seedu/addressbook/commands/Command.java]: Do you think the `editEmploymentInfo` should logically be an action of general `Command` or `EditEmploymentStatusCommand`? Where do you think it should be put? :)
